tests/tensor_search/test_multimodal_tensor_combination.py

```diff
@@ -0,0 +1,882 @@
+
+        mock_multimodal_combination = unittest.mock.MagicMock()
+        mock_multimodal_combination.side_effect = pass_through_multimodal
+
+        @unittest.mock.patch("marqo.tensor_search.tensor_search.vectorise_multimodal_combination_field",
+                             mock_multimodal_combination)
+        def run():
+            tensor_search.add_documents(config=self.config, index_name=self.index_name_1, docs=[
+                {
+                    "combo_text_image": {
+                        "text_field": "A rider is riding a horse jumping over the barrier.",
+                        "image_field": "https://raw.githubusercontent.com/marqo-ai/marqo/mainline/examples/ImageSearchGuide/data/image1.jpg",
+                    },
+                    "_id": "123",
+                },
+
+                {
+                    "combo_text_image": {
+                        "text_field" : "test-text-two.",
+                        "image_field":"https://raw.githubusercontent.com/marqo-ai/marqo/mainline/examples/ImageSearchGuide/data/image2.jpg",
+                    },
+                    "_id": "234",
+                },
+
+                {  # a normal doc
+                    "combo_text_image_test": "https://raw.githubusercontent.com/marqo-ai/marqo/mainline/examples/ImageSearchGuide/data/image2.jpg",
+                    "_id": "534",
+                }
+            ], mappings = {"combo_text_image" : {"type":"multimodal_combination",
+            "weights":{"image_field": 0.5,"text_field": 0.5}}}, auto_refresh=True)
+
+            # first multimodal-doc
+            real_fied_0, field_content_0 = [call_args for call_args, call_kwargs
+                                            in mock_multimodal_combination.call_args_list][0][0:2]
+            assert real_fied_0 == "combo_text_image"
+            assert field_content_0 ==    {
+                        "text_field": "A rider is riding a horse jumping over the barrier.",
+                        "image_field": "https://raw.githubusercontent.com/marqo-ai/marqo/mainline/examples/ImageSearchGuide/data/image1.jpg",
+                    }
+
+            # second multimodal=doc
+            real_fied_1, field_content_1 = [call_args for call_args, call_kwargs
+                                            in mock_multimodal_combination.call_args_list][1][0:2]
+            assert real_fied_1 == "combo_text_image"
+            assert field_content_1 =={
+                        "text_field" : "test-text-two.",
+                        "image_field":"https://raw.githubusercontent.com/marqo-ai/marqo/mainline/examples/ImageSearchGuide/data/image2.jpg",
+                    }
+
+            # ensure we only call multimodal-combination twice
+            assert len(mock_multimodal_combination.call_args_list) == 2
+
+            return True
+
+        assert run()
+
+    def test_multimodal_field_content_dictionary_validation(self):
```

Comment:
You can also test if well formatted dicts _can_ be inserted as expected (only the bad dicts should not be found).
1. Check this from OpenSearch
2. Check if the add_documents call returns an appropriate failure message (`add_docs_res = tensors_search.add_documents(...`)

Reply:
done